internal/services/manageddevopspools/managed_devops_pool_resource.go

```diff
@@ -0,0 +1,679 @@
+	ResourceGroupName              string                                `tfschema:"resource_group_name"`
+	Tags                           map[string]string                     `tfschema:"tags"`
+	StatefulAgentProfile           []StatefulAgentProfileModel           `tfschema:"stateful_agent_profile"`
+	StatelessAgentProfile          []StatelessAgentProfileModel          `tfschema:"stateless_agent_profile"`
+}
+
+func (ManagedDevOpsPoolResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-.]*[a-zA-Z0-9-]$`),
+				"`name` can only include alphanumeric characters, periods (.) and hyphens (-). It must also start with alphanumeric characters and cannot end with periods (.).",
+			),
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
+		"location":            commonschema.Location(),
+		"stateful_agent_profile": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"grace_period_time_span": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Default:      "00:00:00",
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"max_agent_lifetime": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Default:      "7.00:00:00",
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"manual_resource_predictions_profile":    manualResourcePredictionsProfileSchema("stateful_agent_profile.0"),
+					"automatic_resource_predictions_profile": automaticResourcePredictionsProfileSchema("stateful_agent_profile.0"),
+				},
+			},
+			ExactlyOneOf: []string{"stateful_agent_profile", "stateless_agent_profile"},
+		},
+		"stateless_agent_profile": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"manual_resource_predictions_profile":    manualResourcePredictionsProfileSchema("stateless_agent_profile.0"),
+					"automatic_resource_predictions_profile": automaticResourcePredictionsProfileSchema("stateless_agent_profile.0"),
+				},
+			},
+			ExactlyOneOf: []string{"stateful_agent_profile", "stateless_agent_profile"},
+		},
+		"dev_center_project_resource_id": commonschema.ResourceIDReferenceRequired(&projects.ProjectId{}),
+		"vmss_fabric_profile": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"image": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"aliases": {
+									Type:     pluginsdk.TypeSet,
```

Comment:
Any reason to use `TypeSet` here? Can we either test or confirm with the service team that whether the order of `aliases` will be conserved, in which case we shall prefer using `TypeList`?